models/sap__0employee_attr.sql

```diff
@@ -0,0 +1,105 @@
+with pa0000 as (
```

Comment:
Upon looking into the data it does seem that there is either an issue in the joins within this model, or within the date change intermediate model.

Here is an example when looking at the staging model where the join in the downstream results in duplicate records (the begda and enda values were pulled from the employee_date_changes intermediate moel)

 - test failure results
![image](https://github.com/fivetran/dbt_sap/assets/74217849/00af2507-110c-45b4-87a4-519bea336999)


 - my investigation into the staging model

```sql
select pernr, begda, endda, trfst
from dbt_joe_sap.stg_sap__pa0008
where pernr = 00100003
    and begda < 19971231
    and endda > 19970801
order by begda;
```

![image](https://github.com/fivetran/dbt_sap/assets/74217849/ba8e5d6b-d619-4400-a641-0f5a7946dd73)

Reply:
Yes, this is an error I discovered within our internal data. 

As you can see, this record 3 has an overlapping date range with two other records, which shouldn't be possible. 

<img width="679" alt="Screenshot 2023-07-06 at 2 58 30 PM" src="https://github.com/fivetran/dbt_sap/assets/108772760/add4aed4-3412-4792-9324-19068a3a4093">

Trfst is pay scale level, which should be not null for as long as the employee is with the company. Since there is an overlapping date range with a pay scale level for this employee number, my suspicion is this record is erroneous and probably should be removed from our SAP internal data.